File: leetcodejs/array/01two_sum.js
function twoSum(nums, target) {
  var tmp = {};
  for (var i = 0; i < nums.length; i++) {
    var tmp1 = target - nums[i];
    if (tmp1 in tmp) {
      var tmp2 = tmp[tmp1];
      return [tmp2, i];
    }
    tmp[nums[i]] = i;
  }
  return null;
}
var nums1 = [2, 7, 11, 15];
var target1 = 9;
console.log(twoSum(nums1, target1));

File: leetcodepy/array/01two_sum.py
def two_sum(nums, target):
    tmp = {}
    for i, num in enumerate(nums):
        tmp1 = target - num
        if tmp1 in tmp:
            return [tmp[tmp1], i]
        tmp[num] = i
    return None

if __name__ == '__main__':
    nums = [1, 2, 3, 4, 5]
    target = 10
    print(two_sum(nums, target))
